pyanaconda/modules/payloads/payload/rpm_ostree/rpm_ostree.py

```diff
@@ -96,32 +104,47 @@ def setup_kickstart(self, data):
     def _get_ostree_source(self):
         """Get source for RPM OSTree.
 
-        Find out if we need OSTree repo or container source type.
+        Find out if we need OSTree repo, container or bootc source type.
         """
-        return self._get_source(SourceType.RPM_OSTREE_CONTAINER) or \
+        return self._get_source(SourceType.BOOTC) or \
+            self._get_source(SourceType.RPM_OSTREE_CONTAINER) or \
             self._get_source(SourceType.RPM_OSTREE)
 
-    def install_with_tasks(self):
-        """Install the payload.
+    def _install_with_tasks_bootc(self, data):
+        """Create the set of tasks to install the payload with bootc
 
-        :return: list of tasks
+        Similar to ostree flow, but bootc handles bootloader setup, so we skip
+        CopyBootloaderDataTask. DeployBootcTask handles deployment using physroot,
+        then SetSystemRootTask sets the system root to the deployment path,
+        then PrepareBootcMountTargetsTask sets up bind mounts from physroot to sysroot.
         """
-        ostree_source = self._get_ostree_source()
-
-        if not ostree_source:
-            log.debug("No OSTree RPM source is available.")
-            return []
-
-        data = ostree_source.configuration
+        tasks = [
+            DeployBootcTask(
+                data=data,
+                physroot=conf.target.physical_root,
+                sysroot=conf.target.system_root
+            ),
+            SetSystemRootTask(
+                physroot=conf.target.physical_root
+            ),
+            PrepareBootcMountTargetsTask(
```

Comment:
I guess we don't have TearDownBootcMountTargetsTask on purpose at this point, right?

Reply:
Not sure to be honest, this was skipped before I started ammending, I haven't understood the need for this yet.